sklearn/metrics/pairwise.py

```diff
@@ -527,6 +529,54 @@ def manhattan_distances(X, Y=None, sum_over_features=True,
     return D.reshape((-1, X.shape[1]))
 
 
+def _center_and_normalize(X, copy=False):
+    tol = np.finfo(X.dtype).eps * 100
+    # Make implicit copy
+    X = X - np.mean(X, axis=0)
+
+    norms = row_norms(X)
+    non_const_row = np.abs(norms) > tol
+    if non_const_row.all():  # if they are all above treshold
+        X /= norms[:, np.newaxis]
+    elif non_const_row.any() is False:
```

Comment:
change this to ``elif not non_const_row.any()``. Using ``is`` to compare with ``True``/``False`` is not recommended.